### ISSUE TITLE
Add new way to block UI during checkout for payment request buttons

### ISSUE THREAD

**Issue by v18** [woocommerce/woocommerce-gateway-stripe#1539]
Add the ability to block different parts of the UI differently during checkout - one way for the checkout form, and another way for the payment request buttons.

More details here:
https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1509#issuecomment-819803665

This fix should be built on top of this PR:
https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1509

Once the PR for this fix is approved, merge PR #1509 to `trunk`, and then merge the PR for this issue to `trunk`. This should be done in one go so that we can release this feature in the next Stripe release with the correct UI for blocking the checkout.

**Comment by asumaran** [comment]
https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1509#issuecomment-819803665

> We should be able to do both, block/unblock and also disable/enable the Payment Request Button.

@v18 do you mind elaborating this comment? Regarding block/unblock and disable/enable I assume both cases the button behave exactly the same the only difference is that we will hide the spinner for disable/enable events?

**Comment by v18** [comment]
@asumaran - for the payment request button the disable will work the exact same way, but it just won't have a visual spinner showing up. We're trying to avoid this:

<img src="https://user-images.githubusercontent.com/407542/113626538-aaa1d680-9638-11eb-8b1f-9fc8c48cea74.png">

tldr: we still want to disable the payment request button, but not show the spinner

**Comment by asumaran** [comment]
@v18 the spinner is added via [CSS](https://github.com/woocommerce/storefront/blob/trunk/assets/css/base/icons.scss#L599-L616) on the Storefront theme. Switching to another theme the button is displayed correctly when blocked.

This is how it looks using the Twenty One theme: The button is blocked, faded out and has no spinner.

<img width="613" alt="Screen Shot 2021-04-27 at 2 48 50 PM" src="https://user-images.githubusercontent.com/1025173/116303914-afedcf00-a767-11eb-9353-6e4a2718a7fd.png">

What I was thinking we can do is to add a CSS classname to the Payment Request Button according to the state of the button e.g `wc_request_button_is_blocked` when blocked or `wc_request_button_is_disabled` when disabled. This way the current theme will be able to customize the button correctly depending of the button's state.

We already do this but for customized buttons **only**,  I think we should do it for any button type wether the button is custom, default, etc. What do you think?

https://github.com/woocommerce/woocommerce-gateway-stripe/blob/c239804c627f96fc8b5573a149a09461dcecaf74/assets/js/stripe-payment-request.js#L689-L691

**Comment by asumaran** [comment]
@diegocurbelo what do you think on the suggested approach above?

**Comment by diegocurbelo** [comment]
> @diegocurbelo what do you think on the suggested approach above?

I agree with the solution.. it's what we need to unblock #1509.
@asumaran  We also need to create a PR to the Storefront theme to support both blocked and disabled classes, right?

**Comment by asumaran** [comment]
> @asumaran We also need to create a PR to the Storefront theme to support both blocked and disabled classes, right?

I guess so. I can prepare a PR for the Storefront theme as well 👍

**Comment by asumaran** [comment]
Submitted a PR to the Storefront theme to hide the spinner when the Payment Request button is disabled https://github.com/woocommerce/storefront/pull/1672

**Comment by asumaran** [comment]
issue was closed by mistake. reopening
